src/main/java/com/battleships/battleships/BattleshipSquare.java

```diff
@@ -1,13 +1,17 @@
 package com.battleships.battleships;
 
+//inheritance versus aggregation?
 public class BattleshipSquare {
 
     private String type;
+    //
```

Comment:
It feels like `state`, `type` and `destroyedType` are all doing similar things, are they really separate fields?